definitions/tools/samtools_sort.cwl

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env cwl-runner
```

Comment:
Is there a reason to prefer this over the mark_duplicates_and_sort.cwl command already in the tree? Can we keep marking duplicates in this workflow without causing issues downstream? Stringtie happily ignores those duplicate marks, I know...

Reply:
was looking at ` mark_duplicates_and_sort.cwl` and I think that should work AFAIK

Reply:
well no, spoke to soon markdup expects `ASSUME_SORT_ORDER=coordinate` where as the bam that is output from star_align is unsorted. [markdup sort order](https://github.com/samtools/htsjdk/issues/792#issuecomment-275956015)

Reply:
You can pass the input as unsorted
https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.4.0/picard_sam_markduplicates_MarkDuplicates.php#--ASSUME_SORT_ORDER

Reply:
That's about to be fixed in the PR that Sai just made: https://github.com/genome/analysis-workflows/pull/678

Reply:
picard currently does not handle unsorted bams, which is default output from star (see referenced issue above) i see similar error will keep sort_bam step as is, add mark_dup post sort_bam step

Reply:
added mark_dup step

Reply:
I'm still confused about this - Samtools sort and picard sort have some subtle differences in output order (which is crazy, but true).  In general, using picard seems to be the preferred method, because some of the picard tools may not operated as expected when given samtools sorted bams.  

It's fine to merge in this cwl, as someone may need samtools sort at some point. If there's not a strong reason to prefer samtools sort here, I think I'd suggest using picard to sort, as implemented in the mark_duplicates_and_sort cwl.  Should be a pretty quick change to swap that in